### ISSUE TITLE
Can we use this without running geth ??

### ISSUE THREAD

**Issue by Adonis2115** [carsenk/explorer#3]
Hey I want to know can we use this without running geth, like with testrpc or IP address running blockchain on other server and if so how could I do that. I tried to change few things but could not make it work.

Really Appreciate your work.

**Comment by metaspartan** [comment]
This explorer was built with the intentions of utilizing it with geth. It uses https://github.com/ethereum/web3.js/ for the RPC calls. You could change from another different geth implementation, but its current development track is to work strictly with geth and the web3.js.

**Comment by stone212** [comment]
@carsenk: is there anything to install for web3?  I am not able to run this (see bug/question on explorer page:
https://github.com/etherparty/explorer/issues/26

And I am thinking that maybe web3 is not configured correctly for me.  Is that a possibility?  I don't know anything about web3, but I *think* it is project specific, so it should install when npm install does its thing.  Yes?  

Thank you!

**Comment by reiven** [comment]
@Adonis2115 just FYI i'm currently using this explorer with parity instead of geth and works like a charm

**Comment by stone212** [comment]
@reiven How did you install web3?  Generally I install as the user running Parity.  But the explorer doesn't work for me.  Did you install web3 in a different way?

**Comment by stone212** [comment]
@reiven Also did you use the --geth emulation flag for Parity?  Thank you.

**Comment by reiven** [comment]
hey @stone212 i've runned parity without the --geth flag, just started and thats it. I didnt install web3 at all, just set the required flags to let the explorer comunicate via RPC.
Here [is the gist](https://gist.github.com/reiven/aa175b7eec3c68e1aa49e0f47dd3805d) to run parity+explorer inside a docker-compose

**Comment by stone212** [comment]
@reiven I'm not a Docker user but thank you.  Can you just send the actual command line options or config file you use to invoke Parity when explorer works?

**Comment by stone212** [comment]
@reiven Actually can you send the command line options, your firewall settings, reverse proxy settings, and also a copy of your /app/app.js file and any other files that you modified?  I know that is a lot to ask but it will be useful to everyone who is trying to work with Parity and this explorer.

**Comment by reiven** [comment]
@stone212 as you can see in the docker files, i run parity this way:

> parity --jsonrpc-interface all --jsonrpc-hosts all --jsonrpc-cors * --dapps-interface 0.0.0.0 --dapps-hosts all --ui-interface 0.0.0.0 --ui-no-validation --chain dev

For the Fiwall settings, i've just opened TCP ports 8545/8080/8180
And i didnt modified anything in the parity node itself

**Comment by stone212** [comment]
@reiven What about the /app/app.js file?  Have you modified that?  Because I do not have any luck using the settings you give above.

**Comment by reiven** [comment]
No, didnt modify it at all. Which version of parity are you trying?

**Comment by reiven** [comment]
hey @stone212 i see you also have a ticket opened in the parity repo, looks like the problem was using v1.7.x , and the answer fixed my problems with docker. So i can confirm you that i was able to run the explorer with parity v1.7.0 (stable) using docker , without any modification

**Comment by stone212** [comment]
@reiven Hi.  Okay, so you said "For the Fiwall settings, i've just opened TCP ports 8545/8080/8180
And i didnt modified anything in the parity node itself"  But that's impossible!  The explorer runs on port 8000.  I don't even know what port 8080 is.  How can you run without port 8000 open?

And should I do something with port 8080?  I did open it now and there is no change.  I wonder if I must reverse proxy it because I am running remotely?  No.  The reverse proxy should only be for the explorer UI I think.

> No, didnt modify it at all.

You did not modify the /app/app.js file?  Line 6 of that file reads:

`var GETH_HOSTNAME       = "localhost";  // put your IP address!`

Are you viewing the explorer remotely?

> Which version of parity are you trying?

1.6.0, 1.6.10, 1.7.2, 1.8.0.  The result is always the same.  I do not think it is connected to the other ticket (in the parity repo) because I had the problems with the explorer going back much much longer.

**Comment by reiven** [comment]
oh, correct i modify that from the dockerfile

```FROM node:alpine

WORKDIR /app

RUN apk add --no-cache make gcc g++ git bash
RUN git clone https://github.com/carsenk/explorer.git
RUN cd explorer && npm install

WORKDIR /app/explorer
RUN sed -ie 's/localhost/0.0.0.0/g' package.json

EXPOSE 8000

CMD npm start```

**Comment by stone212** [comment]
This now works for me in Parity with an upgrade to 1.7.6.

This does the trick:

<code>
$ parity --jsonrpc-hosts="[public-ip-address]" 
</code>

You also have to put your IP in api.json
